libchisel/src/lib.rs

```diff
@@ -37,6 +37,12 @@ pub trait ModuleValidator {
     fn validate(&self, module: &Module) -> Result<bool, ModuleError>;
 }
 
+pub trait ModulePreset {
+    fn with_preset(&str) -> Result<Self, ()>
```

Comment:
I think we could have an `enum` for settings, which is then extended by the modules, just like how `ModuleError` works.

Reply:
Can you give an example of such a setting enum? I think that instantiating modules with the preset as a string is more robust when considering the possibility of third party modules as well as nonstandard presets.

Reply:
`enum` for the optional settings, such as `payload` in `deployer`.

Reply:
hmm. I'm starting to think that once configuration becomes a bit complex we should have a dedicated context structure